program/rust/src/rust_oracle.rs

```diff
@@ -0,0 +1,33 @@
+use super::c_entrypoint;
+use super::c_oracle_header;
+use solana_program::pubkey::Pubkey;
+use solana_program::sysvar::slot_history::AccountInfo;
+
+///Calls the c oracle update_price, and updates the Time Machine if needed
+pub fn update_price(
```

Comment:
What about having this and `update_version` return a `ProgramResult` for better error handling.

Reply:
`pub extern "C" fn entrypoint(input: *mut u8) -> u64 ` does not return program resunt, and neither does c_entry_point.

We would have to do something similar to entrypoint! to make the first work, and we would have to wrap c_entrypooint with something so that it also does. IDK how worth it is, I think we should prioritize having an MVP to test, what do you think?

Reply:
I would probably solve this by returning a `Result<u64, OracleError/ProgramError>`